presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarController.java

```diff
@@ -0,0 +1,273 @@
+        this.configDeployer = requireNonNull(configDeployer);
+    }
+
+    public void installAsync(URL packageUrl, boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        if (!checkDependencies) {
+            throw new PrestoManagerException("Unsupported parameter 'checkDependencies' for tarball installation");
+        }
+        Path tempFile = getTarPackage(packageUrl);
+        try {
+            tarInstall(tempFile);
+            configDeployer.deployDefaultConfig(configDir, dataDir, pluginDir);
+            configDeployer.deployDefaultConnectors(catalogDir);
+            postInstall();
+            LOGGER.debug("Successfully installed Presto");
+        }
+        finally {
+            deleteTempFile(tempFile);
+        }
+    }
+
+    private Path getTarPackage(URL packageUrl)
+            throws PrestoManagerException
+    {
+        Path tempFile;
+        try {
+            LOGGER.debug("Downloading file from url: %s", packageUrl);
+            tempFile = downloadFile(packageUrl);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException(format("Failed to download file: %s", packageUrl.toString()), e);
+        }
+        return tempFile;
+    }
+
+    private void tarInstall(Path tarFile)
+            throws PrestoManagerException
+    {
+        try {
+            if (isDirectory(INSTALLATION_DIR)) {
+                throw new PrestoManagerException(format("Directory '%s' already exists", INSTALLATION_DIR.toString()));
+            }
+            else {
+                createDirectories(INSTALLATION_DIR);
+            }
+            createDirectories(configDir);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to create directory", e);
+        }
+        int untarResult = executor.runCommand("tar", "-xzvf", tarFile.toAbsolutePath().toString(),
+                "-C", INSTALLATION_DIR.toAbsolutePath().toString());
+        if (untarResult != 0) {
+            throw new PrestoManagerException("Failed to install Presto", untarResult);
+        }
+    }
+
+    private void postInstall()
+            throws PrestoManagerException
+    {
+        try {
+            Path prestoDir = Files.list(INSTALLATION_DIR)
+                    .filter(path -> isDirectory(path))
+                    .reduce((a, b) -> {
+                        throw new IllegalStateException(format("Multiple directories within `%s`", INSTALLATION_DIR));
+                    })
+                    .get().getFileName();
+            launcherScript = Optional.of(INSTALLATION_DIR.toAbsolutePath().resolve(prestoDir).resolve("bin/launcher"));
+            updateProperty(configDir.toAbsolutePath().resolve("node.properties"), "plugin.dir",
+                    INSTALLATION_DIR.toAbsolutePath().resolve(prestoDir).resolve("plugin").toString());
+            /**
+             * If there are extra spaces in this string, there will be problems
```

Comment:
I split the launcherConfig in `runLauncherCommand` so extra spaces would create empty arguments and `processBuilder` will throw an error while executing the command